ziti/cmd/ascode/download/download.go

```diff
@@ -0,0 +1,407 @@
+
+	cmd := &cobra.Command{
+		Use:   "export [entity]",
+		Short: "Export Ziti entities",
+		Long: "Export all or selected Ziti entities.\n" +
+			"Valid entities are: [all|ca/certificate-authority|identity|edge-router|service|config|config-type|service-policy|edgerouter-policy|service-edgerouter-policy|external-jwt-signer|auth-policy|posture-check] (default all)",
+		Args: cobra.MinimumNArgs(0),
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			err := d.Init(out)
+			if err != nil {
+				panic(err)
+			}
+		},
+		Run: func(cmd *cobra.Command, args []string) {
+			err := d.Execute(args)
+			if err != nil {
+				panic(err)
+			}
+		},
+		Hidden: true,
+	}
+
+	v := viper.New()
+
+	// When we bind flags to environment variables expect that the
+	// environment variables are prefixed, d.g. a flag like --number
+	// binds to an environment variable STING_NUMBER. This helps
+	// avoid conflicts.
+	viper.SetEnvPrefix(c.ZITI) // All env vars we seek will be prefixed with "ZITI_"
+
+	// Environment variables can't have dashes in them, so bind them to their equivalent
+	// keys with underscores, d.g. --favorite-color to STING_FAVORITE_COLOR
+	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
+
+	v.AutomaticEnv()
+
+	cmd.Flags().SetInterspersed(true)
+	cmd.Flags().BoolVar(&d.ofJson, "json", true, "Output in JSON")
+	cmd.Flags().BoolVar(&d.ofYaml, "yaml", false, "Output in YAML")
+	cmd.MarkFlagsMutuallyExclusive("json", "yaml")
+
+	cmd.Flags().StringVarP(&d.filename, "output-file", "o", "", "Write output to local file")
+
+	edge.AddLoginFlags(cmd, &d.loginOpts)
+	d.loginOpts.Out = out
+	d.loginOpts.Err = errOut
+
+	return cmd
+}
+
+func (d *Download) Init(out io.Writer) error {
+	d.verbose = d.loginOpts.Verbose
+
+	logLvl := logrus.InfoLevel
+	if d.verbose {
+		logLvl = logrus.DebugLevel
+	}
+
+	pfxlog.GlobalInit(logLvl, pfxlog.DefaultOptions().Color())
+	internal.ConfigureLogFormat(logLvl)
+
+	client, err := mgmt.NewClient()
+	if err != nil {
+		loginErr := d.loginOpts.Run()
+		if loginErr != nil {
+			log.Fatal(err)
+		}
+		client, err = mgmt.NewClient()
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+	d.client = client
+
+	if d.filename != "" {
+		o, err := NewOutputToFile(d.verbose, d.ofJson, d.ofYaml, d.filename)
+		if err != nil {
+			return err
+		}
+		output = *o
+	} else {
+		o, err := NewOutputToWriter(d.verbose, d.ofJson, d.ofYaml, out)
+		if err != nil {
+			return err
+		}
+		output = *o
+	}
+
+	return nil
+}
+
+func (d *Download) Execute(input []string) error {
+
+	args := arrayutils.Map(input, strings.ToLower)
+
+	d.authPolicyCache = cache.New(cache.NoExpiration, cache.NoExpiration)
+	d.configCache = cache.New(cache.NoExpiration, cache.NoExpiration)
+	d.configTypeCache = cache.New(cache.NoExpiration, cache.NoExpiration)
+	d.externalJwtCache = cache.New(cache.NoExpiration, cache.NoExpiration)
+
+	result := map[string]interface{}{}
+
+	all := slices.Contains(args, "all") || len(args) == 0
+	if all ||
```

Comment:
this whole section is confusing to the 'first time reader' (me). Can we discuss what's going on here with these or's?